### ISSUE TITLE
exhaustive-deps placed in the correct spot?

### ISSUE THREAD

**Issue by donaldpipowitch** [oxc-project/oxc#13311]
It looks like in the recent oxlint version ("1.13.0") the exhaustive-deps diagnostics are reported at a different place.

Previously (dependency array):

```tsx
  useEffect(() => {
    emit(EVENTS.CLEAR);
    // oxlint-disable-next-line exhaustive-deps
  }, []);
```

Now (first usage of an unhandled dependency):

```tsx
  useEffect(() => {
    // oxlint-disable-next-line exhaustive-deps
    emit(EVENTS.CLEAR);
  }, []);
```

Was this an intended change? I cannot find a recent discussion about it.

I feel like this is less ergonomic to use and therefor consider it a bug.

If you use other logic earlier inside the hook this comment will be placed somewhere in the middle _on the first usage of an unhandled dep_. I feel this is less refactoring friendly and less visible for longer hooks. (And yes I'm well aware that these comments should be avoided in the first place.)

**Comment by sparebytes** [comment]
I liked the new behavior where the error occurs at the first instance. That gave me a way to ignore certain "dependencies". EG,

```ts
useEffect(() => {
  // oxlint-disable-next-line exhaustive-deps
  void form; 

  // Any variables I want to use without making a dependency are
  // referenced at the top of the callback with oxlint-disable

  form.setFieldValue("email", data.email)
}, [data]);
```

I have no way to selectively ignore certain dependencies if the report is on the dependency array. I would really appreciate if there was some way to keep this behavior.

NOTE: `// oxlint-disable-next-line exhaustive-deps` worked in `1.15.0` but does not in `1.16.0`.

**Comment by Boshen** [comment]
This was fixed by #13781 (merged 2025-09-15), which makes the disable-directive matcher respect the diagnostic's **primary** span.

The `exhaustive-deps` diagnostic emits a secondary label at the dependency usage and a **primary** label at the dependency array. The fix in `Message::new` now prefers the primary label when resolving the span used for `oxlint-disable-next-line`:

```rust
let span = error
    .labels
    .iter()
    .find(|span| span.primary())      // prefer primary (the dep array)
    .or_else(|| error.labels.first())
    .map(|span| Span::new(span.offset(), span.offset() + span.len()))
    .unwrap_or_default();
```

Verified on current `main`:

```jsx
function App({ value }) {
  useEffect(() => {
    doStuff(value);
    // oxlint-disable-next-line react/exhaustive-deps
  }, []); // ✅ suppressed — disable comment at the dep array works again
}
```

The diagnostic header also points back at the dependency array (`test.jsx:N:6` on the `[]`), restoring the pre-1.13.0 ergonomics. Closing as resolved.
